app/src/main/java/com/android/sample/model/ml/FeedbackML.kt

```diff
@@ -0,0 +1,73 @@
+package com.android.sample.model.ml
+
+/**
+ * Abstract base class for feedback
+ *
+ * @property exerciseName The name of the exercise
+ * @property duration Duration in seconds
+ * @property numberOfRepetitions Number of repetitions
+ * @property accuracyScore Percentage of how well it was performed
+ */
+abstract class ExerciseFeedback(
+    open val exerciseName: String,
+    open val duration: Double,
+    open val numberOfRepetitions: Int,
+    open val accuracyScore: Double
+)
+
+/**
+ * Data class representing feedback for pose-related exercises.
+ *
+ * @property balanceScore Balance metric for poses like tree pose
+ * @property alignmentFeedback Joint alignment feedback
+ * @property exerciseName The name of the exercise
+ * @property duration Duration in seconds
+ * @property numberOfRepetitions Number of repetitions
+ * @property accuracyScore Percentage of how well it was performed
+ */
+data class PoseFeedback(
+    val balanceScore: Double? = null,
+    val alignmentFeedback: Map<String, String> = emptyMap(),
+    override val exerciseName: String,
+    override val duration: Double,
+    override val numberOfRepetitions: Int,
+    override val accuracyScore: Double
+) : ExerciseFeedback(exerciseName, duration, numberOfRepetitions, accuracyScore)
+
+/**
+ * Data class representing feedback for repetition-based exercises.
+ *
+ * @property angleThreshold Minimum angle for valid repetition
+ * @property jointFeedback Joint-specific feedback
+ * @property exerciseName The name of the exercise
+ * @property duration Duration in seconds
+ * @property numberOfRepetitions Number of repetitions
+ * @property accuracyScore Percentage of how well it was performed
+ */
+data class RepetitionExerciseFeedback(
+    val angleThreshold: Double? = null,
+    val jointFeedback: Map<String, String> = emptyMap(),
```

Comment:
Would there maybe be a way to turn JointFeedback into a dataclass instead of using a Map for it ? If not it's already great like this

Reply:
Yes indeed it is a better idea.